### PR TITLE
Fixed scheduling ID of pipelines

### PR DESCRIPTION
**Pull request by Skarlso**
Deals with #108 

screen shot 2018-09-24 at 07 16 28

### CODE REVIEW

**Comment by michelvocks** [comment]
Your fix does not work, @Skarlso 😞 
This will increment the ID globally. Which means when you have multiple pipelines the IDs will be shared among all pipeline runs e.g. start pipeline A -> will get ID 1; start pipeline B -> will get ID 2 (should get also ID 1). 

This is also incompatible to existing boltDB databases.

**Comment by Skarlso** [comment]
@michelvocks Alright. This time it's working with multiple pipelines and multiple runs with quick button smashes. It's also warding against if multiple users are using Gaia at the same time and are trying to trigger a build at the same time. :) I think you'll be happy with this one. Hopefully. :D

screen shot 2018-09-24 at 22 36 28

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -243,9 +243,34 @@ func (s *Scheduler) schedule() {
...
 // and save it in our store. The scheduler will later pick this up and will continue the work.
 func (s *Scheduler) SchedulePipeline(p *gaia.Pipeline, args []gaia.Argument) (*gaia.PipelineRun, error) {
+
+	// Introduce a semaphore locking here because this function can be called
+	// in parallel if multiple users happen to trigger a pipeline run at the same time.
+	// (or someone is just simply eager and presses (Start Pipeline) in quick successions).
+	// This means that one of the calls will take slightly longer (a couple of nanoseconds)
+	// while the other finishes to save the pipelinerun.
+	// This is to ensure that the highest ID for the next pipeline is calculated properly.
+	if schedulerRunning {
```
Comment (Skarlso) [inline scheduler/scheduler.go]: why did I even put an if here... 🤦‍♂️

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -243,9 +243,34 @@ func (s *Scheduler) schedule() {
...
+		for schedulerRunning {
```
Comment (Skarlso) [inline scheduler/scheduler.go]: Ok, I need to rethink this semaphore with a channel.

**Comment by Skarlso** [comment]
Omg what am I doing. Just use a waitgroup lock unlock. That's what they are for.

**Pull request merged**